src/libsystemd/sd-journal/journal-def.h

```diff
@@ -42,11 +42,9 @@ typedef enum ObjectType {
         _OBJECT_TYPE_MAX
 } ObjectType;
 
-/* Object flags */
+/* Object flags
+ * The per-compression enums are defined in meson.build so that config.h is self-contained */
 enum {
-        OBJECT_COMPRESSED_XZ   = 1 << 0,
-        OBJECT_COMPRESSED_LZ4  = 1 << 1,
-        OBJECT_COMPRESSED_ZSTD = 1 << 2,
```

Comment:
oh, please let's not remove this here. The idea was that journal-def.h was a more or less complete description of the objects that make up a journal file. If you remove these enums here, then parts of the file format will be define in our build scripts. And that is kinda weird.

Everything in this file is *ABI* and the canonical reference definition of the data structures in journa files, hence never remove anything here really.

Hence, please find some other way to use these flags from meson.build. for example, name the flags differently there, and use an assert_cc() somewhere to ensure they match the values here.

Reply:
There were no build issues with having config.h not self-contained, so I can simply move the definitions back to here - I have no strong opinion either way

Reply:
https://github.com/systemd/systemd/pull/23121